src/Shared/PackageManagers/NPMProjectManager.cs

```diff
@@ -278,6 +278,11 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
         /// <inheritdoc />
         /// <remarks>Currently doesn't respect the <paramref name="includePrerelease"/> flag.</remarks>
         public override async Task<PackageMetadata?> GetPackageMetadataAsync(PackageURL purl, bool includePrerelease = false, bool useCache = true)
+        {
+           return await GetNpmPackageMetadataAsync(purl, includePrerelease, useCache);
+        }
+
+        public  async Task<PackageMetadata?> GetNpmPackageMetadataAsync(PackageURL purl, bool includePrerelease = false, bool useCache = true, bool includeRepositoryMetadata = true)
```

Comment:
Tiniest nit: I believe in C# that optional parameters having a default value provided work the same as having an additional overload which does not specify the argument at all - so the overload on lines 281-283 may not be needed.

Reply:
The first method is override of GetPackageMetadataAsync() method used by all package managers. The second method is specific to NpmPackageManager where we want an option to skip collecting repository metadata. Had we only used the second method with default parameter, the existing code using the BaseProjectManager object to fetch the package metadata from "GetPackageMetadataAsync" method will be impacted as the new method is specific to NpmProjectManager,

Reply:
@morended You're right, I did not notice that they are different methods. I admit I am confused though - why are they separate? (If it's because `GetPackageMetadataAsync` implements an interface, it would seem pretty reasonable to put the new optional argument on the existing interface)

Reply:
I thought this argument is Npm specific and all other package Mangers might not need to implement this.

Reply:
@morended When we were adding a new method that was only implemented for npm, that was definitely npm-specific. 

But making a change to an existing interface implementation is significantly less npm-specific. Also I could see this new flag being honored by other implementations as well - if the npm package manager fetches repository info from GitHub, it seems plausible that other BasePackageManager implementations might do the same thing now or in the future.